javascript/javascript3/week2/main.js

```diff
@@ -0,0 +1,81 @@
+//VISUAL PROMISE
+
+// Function to calculate how much each element needs to be moved in x/y directions.
+function modifyCordinate(currentbox, coordinateObj) {
+  const currentX = currentbox.style.left;
+  const currentY = currentbox.style.top;
+  coordinateObj.x =
+    parseInt(coordinateObj.x) - parseInt(currentX.replace(/px/, ""));
+  coordinateObj.y =
+    parseInt(coordinateObj.y) - parseInt(currentY.replace(/px/, ""));
+  return coordinateObj;
+}
+//Visual Promise to translate one by one:
+function translateOneByOne() {
+  moveElement(
+    document.querySelector("li:nth-child(1)"),
+    modifyCordinate(document.querySelector("ul.marks li:nth-child(1)"), {
+      x: 20,
+      y: 300,
+    })
+  )
+    .then(() => {
+      console.log(" 1st Element has been moved");
+    })
+    .then(() =>
+      moveElement(
```

Comment:
Great implementation of a promise chain. Remember that you can place the console.logs on the same level as the `moveElement` calls to improve readability, if you want to 🙂